### PR TITLE
Apns: Use enum for priority

### PR DESCRIPTION
**Pull request by SMillerDev**
Use an enum for APNS priorities

### CODE REVIEW

File: src/Lunr/Vortex/APNS/APNSPayload.php [hunk 1]
```diff
@@ -31,7 +29,7 @@ public function __construct()
     {
         $this->elements = [];
 
-        $this->elements['priority'] = APNSPriority::HIGH;
+        $this->elements['priority'] = APNSPriority::High;
```
Comment (brianstoop) [inline src/Lunr/Vortex/APNS/APNSPayload.php:32]: Are we sure we don't want to set the value in the array?

File: src/Lunr/Vortex/FCM/FCMApnsPayload.php [hunk 2]
```diff
@@ -114,13 +113,14 @@ public function set_mutable_content(bool $mutable): static
      */
     public function set_priority(string $priority): static
     {
-        $priority = strtoupper($priority);
+        $matching = array_filter(
+            APNSPriority::cases(),
+            fn($case) => strtolower($case->name) == strtolower($priority)
+        );
```
Comment (brianstoop) [inline src/Lunr/Vortex/FCM/FCMApnsPayload.php:119]: > It is also possible to use the [defined()](https://www.php.net/manual/en/function.defined.php) and [constant()](https://www.php.net/manual/en/function.constant.php) functions to check for the existence of or read an enum case if the name is obtained dynamically.

See https://www.php.net/manual/en/language.enumerations.basics.php

**Review by brianstoop — CHANGES_REQUESTED** [review]

File: src/Lunr/Vortex/APNS/APNSPayload.php (hunk 1 above)
Comment (SMillerDev) [reply src/Lunr/Vortex/APNS/APNSPayload.php:32]: What's the benefit of _not_ using the enum?

File: src/Lunr/Vortex/FCM/FCMApnsPayload.php (hunk 2 above)
Comment (SMillerDev) [reply src/Lunr/Vortex/FCM/FCMApnsPayload.php:119]: Then you need to get the casing exactly right though. Or we need to reconstruct the string. Not sure what's the better approach.

Comment (brianstoop) [reply src/Lunr/Vortex/FCM/FCMApnsPayload.php:119]: We have a casing format for enums, right? In this case it doesn't matter much but I think using `defined` or `constant` for this is a better practice

Comment (SMillerDev) [reply src/Lunr/Vortex/FCM/FCMApnsPayload.php:119]: > We have a casing format for enums, right?

Does Lunr have that?

**Pull request closed without merging**